### DISCUSSION TITLE
change background download history

### DISCUSSION THREAD

**Discussion by sasik007** [Godiesc/firefox-gx#220] [Q&A] [answered]
as in the topic, where can I change the background in the download history window? keyboard shortcut ctrl+j
I have a white background and I want to change it
![Zrzut ekranu (6)](https://github.com/user-attachments/assets/55cd227c-4285-441f-b30b-17d6042bb479)

**Comment by Godiesc** [comment]
hi, The css used for that window is `ogx_library.css`. If your Firefox has Dark mode that window should display in dark mode too, but seems like is not your case, maybe the Operative System theme has to do with that, but you have all the code for that window in that file.

**Comment by sasik007** [comment]
I managed to change only part of it but I can't find the rest, can you help?
![Zrzut ekranu (7)](https://github.com/user-attachments/assets/64595f1e-08d6-482f-8bc3-a30aee8a53f6)

**Comment by sasik007** [comment]
Hello, I managed to set everything up again.
Today I've been playing all day, I found everything, but I can't find one element.
as in the picture, after opening a new tab, the background color of the address changes, but only in the new tab, when I click anywhere it changes again
![Zrzut ekranu (8)](https://github.com/user-attachments/assets/6a0aa521-e11d-4f6f-bd54-2977353193c9)
![Zrzut ekranu (9)](https://github.com/user-attachments/assets/15d1c269-2f38-4af6-8ee1-73ee683dc310)



and these are my changes
/* Source file made available under Mozilla Public License v. 2.0 See the main repository for updates as well as full license text. 
   https://github.com/Godiesc/firefox-gx */
/* Source file made available under Mozilla Public License v. 2.0 See the main repository for updates as well as full license text. 
   https://github.com/Godiesc/firefox-gx */
:root,
:root[lwtheme],
:root:not([lwtheme]){
  --toolbar-field-background-color: #1b2329  !important;   /* toolbar_field */
 --toolbarbutton-hover-personal:#1b2329 !important;
--border-radius-urlbar-personal: 30px !important;
--toolbar-field-focus-color: white !important; 
--toolbarbutton-icon-fill:green !important; 
--toolbar-bgcolor: #34444f !important;
--lwt-toolbarbutton-icon-fill-attention:#568696 !important;
--chrome-content-separator-color: green !important;
--toolbar-color:yellow !important;
--arrowpanel-color:yellow !important;
 --organizer-color: white !important;
 --organizer-pane-background: black !important;
 --organizer-border-color: red !important;
--urlbarView-separator-color: green !important;  
--lwt-accent-color: black !important;
  --tab-selected-textcolor: red !important;  
   --lwt-frame: #1b2329!important; 
  --arrowpanel-background:  #1b2329 !important; 
  --toolbarbutton-disabled-opacity: 1 !important;
   
  --toolbarbutton-border-radius: 30px !important;                 
#urlbar[open] #urlbar-background {
    border-color: green !important;
    background-color: #34444f !important; 
    

 
}

**Comment by Godiesc** [comment]
hi,
sorry I couldn't answer you previous question, in my country there are a lot of problems with electric energy (blackouts) and I can't work how I would like, I'm trying to solve many issues for the next Firefox update.
The code you share you should be careful to put comments between `/*` and `*/` I don't know if you pasted here wrongly or do you have the code in that way that could cause problems.
When you open a NewTab the urlbar becomes 'focus' and you have this variable to set the color:

`--toolbar-field-focus-background-color: red !important;`

and the code should work in a better way if you put the code( other than the variables) outside the` root{}`, here is your code:

```
/* Source file made available under Mozilla Public License v. 2.0 See the main repository for updates as well as full license text.
https://github.com/Godiesc/firefox-gx */

:root,
:root[lwtheme],
:root:not([lwtheme]){
--toolbar-field-background-color: #1b2329 !important; / toolbar_field */
--toolbarbutton-hover-personal:#1b2329 !important;
--border-radius-urlbar-personal: 30px !important;
--toolbar-field-focus-color: white !important;
--toolbarbutton-icon-fill:green !important;
--toolbar-bgcolor: #34444f !important;
--lwt-toolbarbutton-icon-fill-attention:#568696 !important;
--chrome-content-separator-color: green !important;
--toolbar-color:yellow !important;
--arrowpanel-color:yellow !important;
--organizer-color: white !important;
--organizer-pane-background: black !important;
--organizer-border-color: red !important;
--urlbarView-separator-color: green !important;
--lwt-accent-color: black !important;
--tab-selected-textcolor: red !important;
--lwt-frame: #1b2329!important;
--arrowpanel-background: #1b2329 !important;
--toolbarbutton-disabled-opacity: 1 !important;

--toolbarbutton-border-radius: 30px !important;
--toolbar-field-focus-background-color: red !important;  /* I added this variable */
}

#urlbar[open] #urlbar-background {
   border-color: green !important;
   background-color: #34444f !important;
}
```

**Comment by sasik007** [comment]
sorry for my language, I used a translator. I have it as I pasted it here, so what you sent has already been corrected?

**Reply by Godiesc** [reply]
> sorry for my language, I used a translator. I have it as I pasted it here, so what you sent has already been corrected?

yes, small change but I fixed it a little.

**Comment by sasik007** [comment]
ok thank you, I'll check how it works and I will continue to have fun, it's just a shame that it's hard to find the right entries

**Reply by Godiesc** [reply]
yeah, still now I find new variables that I could use, and its css and you can change in many ways the style of the elements, for example the variable I shared you change the background color when the urlbar is 'focus' but you can use too other code directly in the element like this:

```
#urlbar[focused] #urlbar-background {
  background-color: yellow !important;
}
```

if you like to change the FIrefox UI with CSS you could spend time [learning CSS](https://www.w3schools.com/css/default.asp) in any internet source and with the [Developer console](https://firefox-source-docs.mozilla.org/devtools-user/browser_toolbox/index.html) you can inspect the elements of the browser and play with them. :)

**Comment by sasik007** [comment]
thanks, but I have one more question: is it possible to make the button background round? now I have a background when I hover the mouse but I would like to have a background of "circles" all the time

**Comment by sasik007** [comment]
it would be nice to make the same background for example of bookmarks

**Reply by Godiesc** [reply]
You can change the border radius (more pixels more roundness)
```
#PersonalToolbar,
#nav-bar{
  --toolbarbutton-hover-background: yellow !important;
  --toolbarbutton-border-radius: 10px !important;
}
```

**Comment by sasik007** [comment]
you misunderstood me, I already have it done but I want the background to be there all the time

**Reply by Godiesc** [reply]
okay, I think this could work:
```
/* Nav-bar buttons */

#nav-bar .toolbarbutton-1 {
  &:not([disabled]) {
    &:not(:hover) > :is(.toolbarbutton-icon, .toolbarbutton-text, .toolbarbutton-badge-stack) {
      background-color: green !important;
    }
  }
}

/* Bookmark-bar buttons */

toolbarbutton.bookmark-item:not(.subviewbutton) {
  background-color: green !important;
}
```

**Comment by sasik007** [comment]
You are great, thank you very much, it works.
the only thing missing is that when you move the mouse over the tabs you are not there and on the tools everything is ok.
![Zrzut ekranu (11)](https://github.com/user-attachments/assets/4fce272c-79c0-4e48-9cf8-b3e0266bfd85)

**Comment by sasik007** [comment]
I recorded a video so you know what I mean, it's OK in the tools, but not in the tabs


https://github.com/user-attachments/assets/f89c4a6b-528d-404e-b0ff-c5b746d10dfe

**Reply by Godiesc** [reply]
the hover variable just put into the bookmarksbar ID:

```
#PersonalToolbar {
  --toolbarbutton-hover-background: yellow !important;
}
```

**Comment by sasik007** [comment]
I paste it into my ogx.trick but it doesn't work, can you modify mine and paste it here?

**Comment by sasik007** [comment]
ok, it works, but it changes the color in the tools and I want to have this shadow in the bookmark

**Reply by Godiesc** [reply]
I don't understand you, sorry. you showed me in your video that you don't have the effect when the mouse is over the bookmarks-bar elements so I gave you the code to change that background. maybe the language gap is a problem here.

**Comment by sasik007** [comment]
here where the mouse shows


https://github.com/user-attachments/assets/10c0b037-59de-4300-bbe3-704cc3376c47

**Reply by Godiesc** [reply]
maybe the problem is a previous code `/* Bookmark-bar buttons */`, change a little bit to solve the problem:
```
/* Bookmark-bar buttons */

toolbarbutton.bookmark-item:not(.subviewbutton):not(:hover) {
  background-color: green !important;
}
```

and after that you can change the 'hover' background with the code I gave you:

```
/* New 'hover' background color */

#PersonalToolbar {
  --toolbarbutton-hover-background: yellow !important;
}
```

**Comment by sasik007** [comment]
it's almost ok but now I have a green background 
--toolbarbutton-hover-background: yellow !important;

this only changes in tools and not in bookmarks


https://github.com/user-attachments/assets/37ea7b82-f0d5-4aa3-8425-ef8f8641e628

**Comment by sasik007** [comment]
--toolbarbutton-hover-background: yellow !important;


I need such an entry but for bookmarks this only works for tools

**Comment by Godiesc** [comment]
all the code I gave you set to general colors for you to change them, and the variable I gave you is into `#PersonalToolbar {...}` so you should use it in the way I give you the code:

https://github.com/user-attachments/assets/ade06fd9-514f-4bc4-99a5-14756804a7cc

**Comment by sasik007** [comment]
--toolbarbutton-hover-personal: yellow !important;

I found that only this entry changed it for me, I don't know why

**Reply by Godiesc** [reply]
paste all your code here to see what is the problem.

**Comment by sasik007** [comment]
/* Source file made available under Mozilla Public License v. 2.0 See the main repository for updates as well as full license text.
https://github.com/Godiesc/firefox-gx */

:root,
:root[lwtheme],
:root:not([lwtheme]){
--toolbar-field-background-color: #1b2329 !important; / toolbar_field */
--toolbarbutton-hover-personal:#1b2329  !important;
--border-radius-urlbar-personal: 30px !important;
--toolbar-field-focus-color: white !important;
--toolbarbutton-icon-fill:green !important;
--toolbar-bgcolor: #34444f !important;
--lwt-toolbarbutton-icon-fill-attention:#568696 !important;
--chrome-content-separator-color: green !important;
--toolbar-color:yellow !important;
--arrowpanel-color:yellow !important;
--organizer-color: white !important;
--organizer-pane-background: black !important;
--organizer-border-color: red !important;
--urlbarView-separator-color: green !important;
--lwt-accent-color: black !important;
--tab-selected-textcolor: red !important;
--lwt-frame: #1b2329!important;
--arrowpanel-background: #1b2329 !important;
--toolbarbutton-disabled-opacity: 1 !important;
--toolbarbutton-border-radius: 30px !important;
--toolbar-field-focus-background-color: #1b2329 !important;  /* I added this variable */
--toolbarbutton-hover-background: yellow !important;
}

#urlbar[open] #urlbar-background {
   border-color: green !important;
   background-color: #1b2329  !important;
}


/* Nav-bar buttons */

#nav-bar .toolbarbutton-1 {
  &:not([disabled]) {
    &:not(:hover) > :is(.toolbarbutton-icon, .toolbarbutton-text, .toolbarbutton-badge-stack) {
      background-color:  #1b2329  !important;
    }
  }
}

/* Bookmark-bar buttons */

toolbarbutton.bookmark-item:not(.subviewbutton):not(:hover) {
  background-color: #1b2329 !important;
}
/* New 'hover' background color */

#PersonalToolbar {
  --toolbarbutton-hover-background: green !important;
}

**Comment by sasik007** [comment]
I think it works, can you check if everything is OK and correct it if necessary?
Thank you very much indeed

**Comment by sasik007** [comment]
could you add a new tab shape?
like the ones I have in my bookmarks??
please

**Reply by Godiesc** [reply]
Maybe you want wavefox: https://github.com/QNetITQ/WaveFox

**Comment by sasik007** [comment]
no, I don't like it, I spent too much time to arrange it the way I want, there's really not much left :-) you've probably had enough of me by now.Can you help again?

**Comment by Godiesc** [comment]
If you like the default tabs you don't need this theme, just some little code to add in your userChrome.css

**Comment by sasik007** [comment]
in firefox these tabs are square and i want bubbles, and i like the menu on the left, maybe you're right but i won't be able to do it.so how can you help with what I wrote above?

**Comment by Godiesc** [comment, marked as answer]
Well I guess you should delete all the code from ogx_tabs-bar.css and use this variables: 
```
--tab-border-radius: 12px !important;
--tab-block-margin: 4px !important;
```

And unset the preference in about:config page to remove tab-shapes

**Comment by sasik007** [comment]
Ok, tomorrow I will continue having fun.and is my code correct? do I need to fix something? I would also like to change these 3 buttons to those from mac os close, minimize etc. is it possible?

**Comment by sasik007** [comment]
ok, last two questions and I'll let you rest for now :-)
1.text color in search bar 
because now when I change the color
--toolbar-color: changes the color of everything for me
2. when I listen to music, there is this speaker icon at the top, but it looks bad, it doesn't look clear, how can I remove it or make it clear?

**Comment by sasik007** [comment]
https://github.com/user-attachments/assets/7c6e17c2-351f-4afa-8ebd-82fe275dfc75

nice option, but illegible and it's hard to hit the speaker icon with the mouse

**Comment by sasik007** [comment]
hey and how will you write back to me??

**Comment by Godiesc** [comment]
hi again,
The speaker icon in your setup seems edited in a wrong way, it's not how the theme show it.

![imagen](https://github.com/user-attachments/assets/c980d823-90ec-4bf2-a68a-c34abd3bbb66)

for the url font color you have this variable: 
`--toolbar-field-color: blue !important;`

PD: The theme was made for work well in Firefox release version, if you use beta/developer/nightly versions you will see issues until I fix them.

**Comment by sasik007** [comment]
I use Mercury

**Comment by sasik007** [comment]
May I ask you not to close this topic? I would like to have more questions and could you answer me in your free time?

**Comment by sasik007** [comment]
![Zrzut ekranu (1)](https://github.com/user-attachments/assets/5bcbba69-0261-4f4b-88cd-8cecc39092a7)


Can anything be changed here? e.g. the shape of these squares, color, etc.?

**Reply by Godiesc** [reply]
for the new-tab page you have the file: [chrome/components/ogx_UC-newtabpage.css](https://github.com/Godiesc/firefox-gx/blob/main/chrome/components/ogx_UC-newtabpage.css), I don't see the wordmark near to the Firefox logo, maybe a difference for your browser and Firefox.

As you can read in the main github page you have the file: [ogx_root-personal.css](https://github.com/Godiesc/firefox-gx/blob/main/chrome/components/ogx_root-personal.css) with usefull rules to change the main aspects of the browser, in that file you have the variable: `--newtab-pinnedsites-background` to change the background color of pinned sites.

**Comment by sasik007** [comment]
I can't find anything to change on this home page.

But a quick question, the entry from this frame when searching is?

![obraz](https://github.com/user-attachments/assets/afbabc77-41b4-469d-bf7b-916de8f1b212)

**Reply by Godiesc** [reply]
you have:
`#urlbar{...}`  and `#urlbar-background{...}`

**Comment by sasik007** [comment]
I can't find it, can you tell me all the paths to it?

**Reply by Godiesc** [reply]
hi, you ask me for the ID, but if you want all the code applied for that section you have all the css files in `chrome/components` folder, the folders have a intuitive name for your understanding. for the section you asked is the file: [ogx_urlbar-searchbar.css](https://github.com/Godiesc/firefox-gx/blob/main/chrome/components/ogx_urlbar-searchbar.css)
